activerecord/test/cases/associations/has_one_associations_test.rb

```diff
@@ -37,8 +37,8 @@ def test_has_one_does_not_use_order_by
     ActiveRecord::SQLCounter.clear_log
     companies(:first_firm).account
   ensure
-    log_all = ActiveRecord::SQLCounter.log_all
-    assert log_all.all? { |sql| /order by/i !~ sql }, "ORDER BY was used in the query: #{log_all}"
+    log_last = ActiveRecord::SQLCounter.log.last
+    assert_no_match(/order by/i, log_last, "ORDER BY was used in the query :#{log_last}")
```

Comment:
Show the failed queries `"ORDER BY was used in the query: #{log_all}"` was added at 3d1ff79 to ease to find the cause.
Can you add the same change to `test_belongs_to_does_not_use_order_by` too?